### PR TITLE
feat: SegmentedControl UIKit implementation

### PR DESCRIPTION
**Pull request by daniel-dumortier**
## Changes description
Implementation of Vitamin tab component for UIKit.
It is a subclass of standard UISegmentedControl, with only customization of: 
- text style and color
- shadow

The PR also contains a `VitaminTabsController` that handles a common use case : display a `VitaminTabs` at the top of the screen, and a viewController below, depending on the tab selected

## Context
UIKit part of #41 

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on an iPhone device/simulator.
- [x] I have tested on an iPad device/simulator.
- [x] If it includes design changes, please ask for a review `design-system-core-team-design` GitHub team.

## Does this introduce a breaking change?

- No

#### iPhone

Simulator Screen Shot - iPhone 8 - 2023-02-06 at 23 22 55

#### iPad

Simulator Screen Shot - iPad (9th generation) - 2023-02-06 at 23 23 59

## Other information

I named the component `VitaminTab`, but I wonder if it should not been named `VitaminSegmentedControl`, that would be more understandable by iOS developers.

### CODE REVIEW

**Review by florentlotthepro — APPROVED** [review]

**Comment by SimonLeclercq** [comment]
Hey @daniel-dumortier, Feel free to discuss with @florentlotthepro about the component name. Keep in mind that we want to stay as close as possible to the developer's experience of the platform. On the Design side, this component is named `segmented control`.

**Comment by SimonLeclercq** [comment]
@daniel-dumortier I'm not able to find the bitrise link 🤔 Can you help me? 😎

**Comment by daniel-dumortier** [comment]
@SimonLeclercq the showcase has not been built yet, that's why you can't find the link ;)
Since I added a new feature on this branch following @florentlotthepro 's suggestion (in short, a generic screen that handles the display of other sub-screens depending on the segment selected), I wait for other code owners technical review before submitting it to you (to ensure it is clean on a code point of view)

File: Sources/VitaminUIKit/Components/Tabs/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+### Common use case
+A common use case of `VitaminTabs` is to use it at the top of a view, and display a viewController below, depending on the selected tab.
+In order to ease this use case, Vitamin provides a `VitaminTabsController`.
+
+You can only instantiate it by ocde (not in a storyboard or xib), and the constructor takes the following parameters:
+- an array of `UIViewController`: the list of controllers you want to display below the `VitaminTabs`. Each view controller should have a `title`, because this title will be displayed as value for each tab
+- a `defaultTabIndex` `Int`: represents the tab index that will be selected at first display
+- a boolean `fullWidthTabs`: indicates if the tabs (that will always be centered horizontally in view) should take the width of the view (with a padding of 10px on each side), or should juste take the needed space for displaying the labels. Default value is false
+- a `tabsTopMargin` `CGFloat`: represents the space between the top of the safe area and th top of the `VitaminTabs`. Default value is 10
```
Comment (florentlotthepro) [inline Sources/VitaminUIKit/Components/Tabs/README.md:33]:
```suggestion
- a `tabsTopMargin` `CGFloat`: represents the space between the top of the safe area and the top of the `VitaminTabs`. Default value is 10
```

File: Sources/VitaminUIKit/Components/Tabs/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- a `tabsBottomMargin` `CGFloat`: represents the space between the bottom of the `VitaminTabs` and the view that will host the viewControllers. Default value is 10
+
+Each of these parameters, except `defaultTabIndex` which has no sens after initialization, can be set afterwards.
+
+⚠️ if you want to modify the list of viewControllers, please do not modify the list directly, but recreate a new list, et reset it to the `viewControllers` property.
```
Comment (florentlotthepro) [inline Sources/VitaminUIKit/Components/Tabs/README.md:38]:
```suggestion
⚠️ if you want to modify the list of `viewControllers`, please do not modify the list directly, but recreate a new list, and reset it to the `viewControllers` property.
```

File: Sources/VitaminUIKit/Components/Tabs/VitaminTabsController.swift [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        }
+    }
+    /// Margin between the bottom of the `VitaminTabs` and the top of the view dsplaying the sub viewControllers
+    /// Default value is 10
+    public var tabsBottomMargin: CGFloat = 10.0 {
+        didSet {
+            tabsBottomMarginConstraint.constant = tabsBottomMargin
+        }
+    }
+    /// Boolean indicting if the `VitaminTabs`should take the full width of the screen
```
Comment (florentlotthepro) [inline Sources/VitaminUIKit/Components/Tabs/VitaminTabsController.swift:35]:
```suggestion
    /// Boolean indicting if the `VitaminTabs` should take the full width of the screen
```

File: Sources/VitaminUIKit/Components/Tabs/VitaminTabsController.swift [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        self.tabViewControllers = viewControllers
+        self.defaultTabIndex = defaultTabIndex
+        self.fullWidthTabs = fullWidthTabs
+        self.tabsTopMargin = tabsTopMargin
+        self.tabsBottomMargin = tabsTopMargin
+    }
+
+    public required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
```
Comment (florentlotthepro) [inline Sources/VitaminUIKit/Components/Tabs/VitaminTabsController.swift:100]: Doesn't it seem to be nice to use it from a storyboard?

**Review by florentlotthepro — COMMENTED** [review]
Like said by Simon, what do you think about `VitaminSegmentedControl`? (and `VitaminSegmentedController`) It's not exactly like Vitamin Android but it feels more iOS. For me, Tabs feels like related to `TabBar`.

File: README.md [hunk 5]
```diff
@@ -88,6 +88,7 @@ The following components are available :
 |Progressbar | Different progressbar styles from the Vitamin Design System. | [UIKit](./Sources/VitaminUIKit/Components/Progressbar#readme)|
 |Snackbar | Snackbar from the Vitamin Design System. | [UIKit](./Sources/VitaminUIKit/Components/Snackbar#readme)|
 |Switch | Swicth from the Vitamin Design System. | [UIKit](./Sources/VitaminUIKit/Components/Switch#readme)|
+|Tabs | Tabs from the Vitamin Design System. | [UIKit](./Sources/VitaminUIKit/Components/Tabs#readme)|
```
Comment (florentlotthepro) [inline README.md:91]:
```suggestion
|Switch | Switch from the Vitamin Design System. | [UIKit](./Sources/VitaminUIKit/Components/Switch#readme)|
|SegmentedControl | SegmentedControl (Tabs) from the Vitamin Design System. | [UIKit](./Sources/VitaminUIKit/Components/SegmentedControl#readme)|
```

**Review by florentlotthepro — APPROVED** [review]
I think that it could be nice to also rename the folder for the component 😇

**Review by florentlotthepro — APPROVED** [review]

**Comment by daniel-dumortier** [comment]
@SimonLeclercq The build is available, you can find the link on Slack

**Review by SimonLeclercq — COMMENTED** [review]
I think I see a slight horizontal shift on the labeltext compared to the container. Can you confirm this? Besides that, it's good for me.

**Comment by daniel-dumortier** [comment]
@SimonLeclercq 
Honestly, I do not really get what you mean by horizontal shift, but for info, I just styled the native SegmentedControl component (with custom font, size, colors, and shadow), thus I do not have any way to shift or move the labels in the segmented control.
I just let iOS place them inside the component.

**Comment by SimonLeclercq** [comment]
Ok so no problem 👍 thanks for feedback !

**Review by SimonLeclercq — APPROVED** [review]

**Pull request merged**